integration-tests/ci-visibility.spec.js

```diff
@@ -44,7 +46,26 @@ const testFrameworks = [
       'ci-visibility/test/ci-visibility-test-2.js'
     ],
     runTestsWithCoverageCommand: './node_modules/nyc/bin/nyc.js -r=text-summary node ./ci-visibility/run-mocha.js',
-    coverageMessage: 'Lines        : 80%'
+    coverageMessage: 'Lines        : 80%',
+    type: 'commonJS'
+  },
+  {
+    name: 'mocha',
+    dependencies: [isOldNode ? 'mocha@9' : 'mocha', 'chai', 'nyc'],
+    testFile: 'ci-visibility/run-mocha.mjs',
+    expectedStdout: '2 passing',
+    extraStdout: 'end event: can add event listeners to mocha',
+    expectedCoverageFiles: [
+      'ci-visibility/run-mocha.mjs',
+      'ci-visibility/test/sum.js',
+      'ci-visibility/test/ci-visibility-test.js',
+      'ci-visibility/test/ci-visibility-test-2.js'
```

Comment:
Probably makes more sense to change all files to mjs for completeness

Reply:
I also don't think we can change the files to mjs if we are running mocha programtically, for those I believe only .js files are supported

Reply:
the test runner could be in esm (run-mocha.mjs) but the rest need to be in js with the current approach

Reply:
I'd assume then just having run-mocha.mjs with all the tests passing should be sufficient? If not then I can change it run using mocha and not node so that we can have .mjs test files

Reply:
let's keep it as is. I can investigate myself how to change it to use mocha instead of node directly. I'm afraid we wouldn't be testing _fully_ esm without the test files also being mjs, but it doesn't have to happen in this PR.